tests/test_validators.py

```diff
@@ -250,3 +250,46 @@ def test_any_of(self):
     def test_none_of(self):
         self.assertEqual(NoneOf(['a', 'b', 'c'])(self.form, DummyField('d')), None)
         self.assertRaises(ValueError, NoneOf(['a', 'b', 'c']), self.form, DummyField('a'))
+
+
+@pytest.mark.parametrize("address", [
+    u"147.230.23.25",
+    u"147.230.23.0",
+    u"127.0.0.1"
+])
+def test_ip4address_passes(address):
+    adr = ip_address()
+    field = DummyField(address)
+    adr(None, field)
+
+
+@pytest.mark.parametrize("address", [
+    u"2001:718:1C01:1111::1111",
+    u"2001:718:1C01:1111::",
+])
+def test_ip6address_passes(address):
+    adr = ip_address(ipv6=True)
+    field = DummyField(address)
+    adr(None, field)
+
+
+@pytest.mark.parametrize("address", [
+    u"2001:718:1C01:1111::1111",
+    u"2001:718:1C01:1111::",
+])
+def test_ip6address_raises(address):
+    adr = ip_address()
+    field = DummyField(address)
+    with pytest.raises(ValidationError):
+        adr(None, field)
+
+
+@pytest.mark.parametrize("address", [
+    u"147.230.1000.25",
+    u"2001:718::::",
+])
+def test_ip4address_raises(address):
+    adr = ip_address(ipv6=True)
+    field = DummyField(address)
+    with pytest.raises(ValidationError):
+        adr(None, field)
```

Comment:
Why are all of these outside the `test_ip_address` function?

Reply:
That's because of the `test_ip_address` is method written in UnitTest style. And @davidism told me,    that I can use the PyTest style for the new tests. Because the original tests should be rewritten to PyTest style too in the future. And so I use the PyTest parametric tests.

Reply:
This is fine.